src/lib.rs

```diff
@@ -1142,20 +1103,23 @@ impl fmt::Show for YieldWaitStrategy {
  * need to also perform a load in order to have acquire semantics. This is why read-modify-write
  * operations are needed, and not just the cheaper load/store operations.
  */
+#[derive(Debug)]
 pub struct BlockingWaitStrategy {
     d: UncheckedUnsafeArc<BlockingWaitStrategyData>,
     // Deep copyable fields that don't need to be shared
     /// Number of times to wait before blocking
-    max_spin_tries_publisher: uint,
+    max_spin_tries_publisher: usize,
     /// Number of times to wait for a consumer sequence before yielding
-    max_spin_tries_consumer: uint,
+    max_spin_tries_consumer: usize,
 }
 
+#[derive(Debug)]
 struct BlockingWaitStrategyData {
     /// True if any tasks are waiting for new slots to be released by the publisher.
     signal_needed: AtomicBool,
     /// Waiting consumers block on, and are signalled by, this.
-    wait_condition: Mutex<Condvar>,
+    mutex: Mutex<()>,
```

Comment:
The way Mutexes and Condvars interact has changed. In the docs, the example uses a `Mutex<bool>`, but I didn't want to move the bool inside a lock, so I left `signal_needed` alone.